### ISSUE TITLE
[BUG] Notifications aren't sent

### ISSUE THREAD

**Issue by stefanelul2000** [tigattack/VeeamDiscordNotifications#18]
**Describe the bug**
Discords notifications are not being sent by all backup jobs.

**To Reproduce**
Setup multiple jobs in Veeam and setup them to send notifications.

**Expected behavior**
Every job should send a notification.

**Relevant part of `log/debug.log`**
Don't have it right now, will post it a bit later.

<img width="392" alt="RemoteDesktopManagerFree_9Ql4yxxE3A" src="https://user-images.githubusercontent.com/42817352/121561038-c81c6380-ca20-11eb-94d2-1bd56c4c4740.png">
This backup sends notifications without any issue.

<img width="427" alt="RemoteDesktopManagerFree_Er5Z4ZssHb" src="https://user-images.githubusercontent.com/42817352/121561098-d4a0bc00-ca20-11eb-9219-8d22167f969e.png">
This backup does not send any notification at all.

**Comment by tigattack** [comment]
Hi @stefanelul2000. Once you've attached logs I can take a look into this for you 👍

**Comment by stefanelul2000** [comment]
[debug.log](https://github.com/tigattack/VeeamDiscordNotifications/files/6641292/debug.log)

I have attached the logs. Hopefully they are the right ones as they do not mention from what Veeam backup job they origiate,

**Comment by stefanelul2000** [comment]
[debug.log](https://github.com/tigattack/VeeamDiscordNotifications/files/6641394/debug.log)

Another job finished just now. This is the latest log for that job

**Comment by stefanelul2000** [comment]
I saw something in the logs. Line 56 in file "DiscordVeeamAlertSender.ps1" needs to be changed to "Import-Module Veeam.Backup.PowerShell" 

Issue is similar to #17 

Edit: That didn't do anything. 
Another important thing to point out is this
"WARNING: This cmdlet is no longer supported for computer backup jobs. Use "Get-VBRComputerBackupJob" instead.
You cannot call a method on a null-valued expression."

**Comment by tigattack** [comment]
Didn't see your update; good thing to sort anyway though. Thought I sorted that one!

So does this line match yours then? [DiscordVeeamAlertSender.ps1#L56](https://github.com/tigattack/VeeamDiscordNotifications/blob/master/DiscordVeeamAlertSender.ps1#L56)  
Do you see anything else in the logs after making that change?

I'm aware of the warning, thanks. A lot of this needs a rewrite, I'm just quite short of time. Progress is happening, just quite slowly 😄

**Comment by stefanelul2000** [comment]
Yeah, the line matches mine now but it still does not work. I am unsure why.

**Comment by tigattack** [comment]
Do you see anything else in the logs after making that change?

**Comment by stefanelul2000** [comment]
The same errors as before. But without the line with the snappin module missing.

**Comment by stefanelul2000** [comment]
[debug.log](https://github.com/tigattack/VeeamDiscordNotifications/files/6642111/debug.log)

This is another log after that change. Still the same errors as before. I am not really sure what is wrong.

**Comment by tigattack** [comment]
Could you try the following and show me the output?

1. Open PowerShell on your Veeam server
2. Run `Import-Module Veeam.Backup.PowerShell -Verbose`

**Comment by stefanelul2000** [comment]
[Import-Module-Verbose.txt](https://github.com/tigattack/VeeamDiscordNotifications/files/6643566/Import-Module-Verbose.txt)

I have attached the output of the command.

**Comment by tigattack** [comment]
Please try replacing your DiscordNotificationBootstrap.ps1 with [this](https://pastebin.com/yZ0290fs) (make sure the filename isn't changed). Trigger a job and please attach the new debug.log once the script has run.
It seems it's failing to get the [session](https://github.com/tigattack/VeeamDiscordNotifications/blob/master/DiscordNotificationBootstrap.ps1#L21) information but this should confirm it.

If that's the case, I'm currently not sure why that would fail, but once it's confirmed I'll see if I can work out what's happening.

**Comment by stefanelul2000** [comment]
I have update it with the provided file. I will report as soon as the next job runs.

**Comment by stefanelul2000** [comment]
[debug.log](https://github.com/tigattack/VeeamDiscordNotifications/files/6660703/debug.log)

I have attached the latest logs.

**Comment by stefanelul2000** [comment]
Hello, do you have any update on this?

**Comment by tigattack** [comment]
Apologies, I've been quite busy lately. I'm hoping to look into this some more ASAP as I'm currently unsure why it's failing.

**Comment by stefanelul2000** [comment]
Ok, sure. Let me know if you have any update.

**Comment by tigattack** [comment]
@stefanelul2000 Apologies for the delay, life has been very busy, but I haven't forgotten about this.

A collaborator and I are investigating the issue and I realised I've not actually confirmed which version you're running.  
Are you running the latest release or have you cloned one of the branches?

**Comment by stefanelul2000** [comment]
@tigattack 
 
I am running on 1.5 I believe with small modifications. The ones mentioned in this ticket.

**Comment by tigattack** [comment]
Great, thanks Stefan.

**Comment by stefanelul2000** [comment]
I have also checked, since 12.06.2021 There haven't been any notifications sent to my discord even thought the script is still enabled on the veeam server.

**Comment by stefanelul2000** [comment]
[veeam-discord-log.txt](https://github.com/tigattack/VeeamDiscordNotifications/files/7224748/veeam-discord-log.txt)

This is the latest log.

**Comment by tigattack** [comment]
Ah, that's very useful actually, seems it's failing to get job and session details. Thanks.

I'll spend some more time on this this weekend and hopefully have something for you to try.

**Comment by stefanelul2000** [comment]
@tigattack Do you have any update for me?

**Comment by tigattack** [comment]
Hi @stefanelul2000, could you please try the [latest release](https://github.com/tigattack/VeeamDiscordNotifications/releases/tag/v2.0-rc1) and let me know if the issue persists?
